client/matching/loadbalancer.go

```diff
@@ -56,6 +56,14 @@ type (
 			taskListType int,
 			forwardedFrom string,
 		) string
+
+		// GetAllPartitions returns the list of all the partitions for a given tasklist
+		// If the taskList is not scalable it will return a empty tasklist
```

Comment:
if the taskList is not *partitioned*,

Reply:
Sorry the comment is misleading, If the tasklist is not partitioned, I am returning array with tasklist name only. I am returning empty list if there is an error finding the domain name.

Reply:
That's weird, whether the task list has 1 or 100 partitions, you should return the same set of result params from this API on success

Reply:
Yes. That's true. Will update.